### PR TITLE
Fix race conditions in unit tests

### PR DESCRIPTION
**Pull request by yjpa7145**
This addresses CUMULUS-1674.

We were seeing intermittent failures in our unit tests, specifically tests that were using Elasticsearch. There were a few reasons for this:

* Some of our tests were deleting all aliases, which meant that test running at the same time would fail.
* Almost all of our tests were using randomly-named ES indexes, but were using the same default alias name. When a test would create a new index and point the default alias at that new index, it would cause other running tests to fail.

This PR makes a number of changes to address those issues, and a couple other related issues that were discovered:

* In addition to randomly-named indexes, the tests now use randomly-named aliases, instead of relying on the default alias
* Tests no longer delete all aliases
* As a result of these changes, the API's serial tests can now be run in parallel, so those have been moved out of the serial directory
* Tests were often operating directly on the index, instead of the alias. Updated tests to perform operations on the alias.
* We were using `process.env.esIndex` in some places, and `process.env.ES_INDEX` in others. Since we are also using `process.env.ES_HOST`, I settled on `process.env.ES_INDEX` and updated all of our tests to use that. `process.env.ES_INDEX` now points at the ES index _or alias_ that our code operates on. Typically, this is going to be an alias. It's called "INDEX" instead of "ALIAS" because the parameter in the ES calls is named `index`, although that value can be an index or an alias.

This PR only affects our tests and not the production code, so there is no Changelog entry.

### CODE REVIEW

File: packages/api/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "cumulus-api": "./bin/cli.js"
   },
   "scripts": {
-    "test": "ava && find ./tests/serial -name '*.js' | xargs ava --serial",
```
Comment (Jkovarik) [inline packages/api/package.json]: Nice.  💯

**Comment by yjpa7145** [comment]
> re: deleting aliases - we don't _need_ to remove them, but did you think about potential edge cases to having aliases not explicitly removed in the tests? I don't see anything in the ES docs that implies any potential issues in this use case

The aliases aren't explicitly deleted, but deleting the index will delete any associated aliases, and we _are_ deleting the indexes after each set of tests.

**Comment by Jkovarik** [comment]
Ah, that makes sense.  I'd seen you can associate an alias with *multiple* indexes, but apparently ES does a better job cleaning up than I'd given it credit for.  Thanks for the explanation.
